runner/context.ts

```diff
@@ -134,26 +140,27 @@ export class Context extends events.EventEmitter {
             resolve();
           }
         });
-        if (maybePromise && maybePromise.then) {
-          // Once typescript ≥1.9 is out I think we'll be able to get rid of
-          // this local variable.
-          const promise = <Promise<any>> maybePromise;
-          promise.then(resolve, reject);
+        if (maybePromise) {
+          maybePromise.then(resolve, reject);
         }
       });
     };
 
-    // We execute the handlers _sequentially_. This may be slower, but it gives us
-    // a lighter cognitive load and more obvious logs.
+    // We execute the handlers _sequentially_. This may be slower, but it gives
+    // us a lighter cognitive load and more obvious logs.
+    await Promise.resolve();
```

Comment:
This was an artifact of debugging that weird error where `done()` throws by more precisely aping the order of operations of the code before it was converted.

Removed.